app/frontend/src/cljs/teet/contract/contract_partners_view.cljs

```diff
@@ -495,15 +506,20 @@
      ["" {:align :right :width "10%"}]
      ["" {:align :right :width "10%"}]]
     (for [employee employees
-          :let [key-person? (get-in employee [:company-contract-employee/key-person?])]]
+          :let [key-person? (get-in employee [:company-contract-employee/key-person?])
+                key-person-status (get-in employee [:company-contract-employee/key-person-status])]]
       [[(str (get-in employee [:company-contract-employee/user :user/family-name]) " "
              (get-in employee [:company-contract-employee/user :user/given-name]))]
        (if (not-empty (:company-contract-employee/role employee))
          [(str/join ", " (mapv #(tr-enum %) (:company-contract-employee/role employee)))]
          [])
        [(if (true? key-person?)
           [:div {:style {:display :flex}}
-           [key-person-icon :gray]]
+           [key-person-icon (cond
+                              (du/enum= key-person-status :key-person.status/approval-requested) :yellow
+                              (du/enum= key-person-status :key-person.status/approved) :green
+                              (du/enum= key-person-status :key-person.status/rejected) :red
+                              :else :gray)]]
```

Comment:
I see the previous implementation handles color keywords, such as `:gray`, but instead of color, we could pass the status keyword to the `key-person-icon` component and act based on that.

Reply:
That is

```
[key-person-icon key-person-status]
```
and then instead of `case`ing by the color, we can `case` (or `cond`) based on the status itself.